tasks/variables.yml

```diff
@@ -49,3 +49,21 @@
     postgresql_unix_socket_directories_mode: >-
       {{ __postgresql_unix_socket_directories_mode | default('02775') }}
   when: postgresql_unix_socket_directories_mode is not defined
+
+- name: Define postgresql_log_dir.
+  set_fact:
+    postgresql_log_dir: "{{ (postgresql_global_config_options | items2dict(key_name='option', value_name='value')).log_directory }}"
+
+- name: Define postgresql_log_dir_is_absolute.
+  set_fact:
+    postgresql_log_dir_is_absolute: '{{ postgresql_log_dir | regex_search("^/.*") != "" }}'
+
+- name: Define postgresql_effective_log_dir, if postgresql_log_dir_is_absolute
+  set_fact:
+    postgresql_effective_log_dir: '{{ postgresql_log_dir }}'
+  when: postgresql_log_dir_is_absolute
+
+- name: Define postgresql_effective_log_dir, if not postgresql_log_dir_is_absolute
+  set_fact:
+    postgresql_effective_log_dir: '{{ postgresql_data_dir }}/{{ postgresql_log_dir }}'
+  when: not postgresql_log_dir_is_absolute
```

Comment:
Quoting @geerlingguy:
> (...) this change seems to add a lot of complexity (...) If this was a few lines or one or two tasks, it would be a lot less stressful to review and merge (...)

I think that this part of the code is the main source of complexity. 

Is it possible to check if the `postgresql_log_dir` path is relative or absolute and in the former case prepend it with `postgresql_data_dir` in a shorter and/or simpler way?

Reply:
I could change
```
    postgresql_effective_log_dir: '{{ postgresql_data_dir }}/{{ postgresql_log_dir }}'
```
to
```
    postgresql_effective_log_dir: '{{ ( postgresql_data_dir, postgresql_log_dir ) | path_join }}'
```
...but this just makes the code slightly more correct (better handle a case where someone would f.e. add a trailing `/` to the `postgresql_log_dir`) but not really much simpler and it would increase the minimum ansible version from [2.8](https://github.com/geerlingguy/ansible-role-postgresql/blob/master/meta/main.yml#L10) to [2.10](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#managing-file-names-and-path-names).

Reply:
I did some change, hope that it helps.